### PR TITLE
implements backend filter functionality for matching fields

### PR DESCRIPTION
**Pull request by LindsayYoung**
Creates and filter function we can extend later, it provides filters when people want a field to provide one or more matches.

Need to check in with @el-mapache about what `filters` should pass to the front end 
Need to add more tests

### CODE REVIEW

File: crt_portal/cts_forms/tests.py [hunk 1]
```diff
@@ -270,24 +270,22 @@ def setUp(self):
...
+    #     self.assertFalse(context['filters'].get('state'))
 
     def test_filter(self):
-        expected_reports = filter(
-            lambda x: x.assigned_section == 'ADM',
-            Report.objects.all()
-        )
+        expected_reports = Report.objects.filter(assigned_section='ADM')
         actual_reports = self.response.context['data_dict']
+        print("\n\nACTUAL-----", actual_reports)
```
Comment (el-mapache) [inline crt_portal/cts_forms/tests.py]: I think we'll want to remove these print statements.

**Comment by el-mapache** [comment]
@LindsayYoung The only filter we need right now is the assigned section functionality, but what you have here is fine.

**Comment by LindsayYoung** [comment]
@el-mapache since you approved, I added an extra test, and removed all print statements, I am going to merge it now

**Comment by el-mapache** [comment]
Great! I'll pull in to my branch in a bit, hopefully it should integrate smoothly.

**Pull request merged**